parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java

```diff
@@ -1063,6 +1068,69 @@ public ColumnChunkPageReadStore readFilteredRowGroup(int blockIndex, RowRanges r
     return internalReadFilteredRowGroup(block, rowRanges, getColumnIndexStore(blockIndex));
   }
 
+  /**
+   * Read data in all parts via either vectored IO or serial IO.
+   * @param allParts all parts to be read.
+   * @param builder used to build chunk list to read the pages for the different columns.
+   * @throws IOException any IOE.
+   */
+  private void readAllPartsVectoredOrNormal(List<ConsecutivePartList> allParts, ChunkListBuilder builder)
+    throws IOException {
+    boolean isVectoredIO = options.useHadoopVectoredIO()
```

Comment:
Move the 3 conditions into a method and make a good document to call out what exact conditions are needed for the user to use vectorized IO. And also we should log it clearly whether or not vectoredIO is used.

Reply:
ok. will log at info with some more details about the request (number of ranges, total amount of data).

We could also report on the total duration of the vectored read...would that be useful?